sql3/parser/ast.go

```diff
@@ -1919,6 +1919,7 @@ func (lit *BoolLit) String() string {
 type DateLit struct {
 	ValuePos Pos       // literal position
 	Value    time.Time // literal value
+	Token    Token     // token found in the sql
```

Comment:
I would add a comment here about how Token is meant to be handled.

```suggestion
	ValuePos Pos       // literal position
	Value    time.Time // literal value

	// Token will be a non-zero value if the date found in the SQL is a built-in
	// literal. This is maintained here in the struct for cases, such as SQL
	// stored for a view, where the SQL string is to be parsed again.
	Token Token
```
or something like that. Maybe I'm not understanding it correctly, but you can describe it however it makes sense.